packages/tracing/src/browser/metrics.ts

```diff
@@ -179,6 +214,63 @@ export class MetricsInstrumentation {
       // Do nothing if the browser doesn't support this API.
     }
   }
+
+  /** Starts tracking the First Input Delay on the current page. */
+  private _trackFID(): void {
+    // Based on reference implementation from https://web.dev/fid/#measure-fid-in-javascript.
+    // Use a try/catch instead of feature detecting `first-input`
+    // support, since some browsers throw when using the new `type` option.
+    // https://bugs.webkit.org/show_bug.cgi?id=209216
+    try {
+      // Keep track of whether (and when) the page was first hidden, see:
+      // https://github.com/w3c/page-visibility/issues/29
+      // NOTE: ideally this check would be performed in the document <head>
+      // to avoid cases where the visibility state changes before this code runs.
+      let firstHiddenTime = document.visibilityState === 'hidden' ? 0 : Infinity;
+      document.addEventListener(
+        'visibilitychange',
+        event => {
+          firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
+        },
+        { once: true },
+      );
```

Comment:
Did you try to share this with the LCP implementation instead of creating multiple event listeners?

Reply:
Done in https://github.com/getsentry/sentry-javascript/pull/2909/commits/c1b72aede1f3de2736acda06615705e1bf63efa2